### ISSUE TITLE
Crank doesn't let you pass query strings

### ISSUE THREAD

**Issue by insulind** [SignalR/SignalR#4020]
### Expected  behavior
- To be able to load test a signalR endpoint that requires query string params passed to it
### Actual behavior
- There is currently no parameter to pass query strings to crank. If the url passed contains a question mark it crashes as there is an unhandled aggregate exception that contains the original exception that says, url can't contain a '?'
### Steps to reproduce
- Pass a url to crank with a query string such as http://mywebservice.com:8080/signalR?param=value and it doesnt work

I needed this functionality for some testing I was doing and implemented it myself. I would love to contribute to project, so if people are happy I would happily make an attempt at this. Let me know!

**Comment by moozzyk** [comment]
We do accept contributions so feel free to send a PR.

**Comment by insulind** [comment]
Cool, I'll get one in asap. Thanks!

**Comment by insulind** [comment]
@moozzyk  I've got the dev branch and built it and I've ran the tests and some are failing. Is that to be expected? (I haven't made my changes fyi)

**Comment by moozzyk** [comment]
You are likely missing some dependencies. Can you show the errors? (I think functional tests needed IIS manager and you may need to install some UWP tooling if it is not installed)

**Comment by insulind** [comment]
```
Test Name:	Microsoft.AspNet.SignalR.Client.Tests.ConnectionFacts+Start.AwaitingOnFailedStartAndThenStoppingDoesntHang(hostType: Memory, transportType: LongPolling, messageBusType: Default)
Test FullName:	Microsoft.AspNet.SignalR.Client.Tests.ConnectionFacts+Start.AwaitingOnFailedStartAndThenStoppingDoesntHang (ae35e591e2a092a0e79470de8f84ec11011e89a3)
Test Source:	D:\SignalR\tests\Microsoft.AspNet.SignalR.FunctionalTests\Server\Connections\ConnectionFacts.cs : line 555
Test Outcome:	Failed
Test Duration:	0:00:21.06

Result StackTrace:	
at Microsoft.AspNet.SignalR.Client.Tests.ConnectionFacts.Start.<AwaitingOnFailedStartAndThenStoppingDoesntHang>d__13.MoveNext() in D:\SignalR\tests\Microsoft.AspNet.SignalR.FunctionalTests\Server\Connections\ConnectionFacts.cs:line 570
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Result Message:	
Assert.True() Failure
Expected: True
Actual:   False


```

Its the 6 'Theories' that use HostType.Memory.

Also 4 tests in Client.Store.Tests, which I assume is because I'm missing some UWP tooling as you mentioned

 Microsoft.AspNet.SignalR.Client.Store.Tests.EndToEndTests:
- WebSocketReconnects
- WebSocketReconnectsIfConnectionLost
- WebSocketSendReceiveTest
- SendingMessageWhenTransportIsReconnecting

**Comment by insulind** [comment]
Hopefully last thing (sorry just want to make sure I haven't broken anything).

When I build, it builds successfully but I get this error

![image](https://user-images.githubusercontent.com/24652224/31300177-27dad7b6-aaea-11e7-89b3-2029b710fb8d.png)

**Comment by moozzyk** [comment]
I don't think you can break the tests you posted by changing Crank. Just submit the PR and I can run tests in my environment and investigate.
For the second issue I think I have seen it and the workaround was to build from command line and then open VS

**Comment by moozzyk** [comment]
This has been merged as 35a44764686f796baf7c4256e5d52ab5f9d1162b
